fiatconnect-api.md

```diff
@@ -391,105 +407,204 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+The `issued-at` field MUST be the ISO 8601 datetime string of the time at which the client generated the message.
+The `expiration-time` field MUST be an ISO 8601 datetime string that specifies when this message will no longer valid, and when a session created with this message
+will expire. This field MUST NOT be more than four hours (14400 seconds) later than the `issued-at` field.
+
+Note that in cases where a client is attempting to authorize themselves as the owner of a contract-owned account (rather than an externally-owned account), the `address`
+field MUST correspond to the address of the on-chain contract account, rather than the externally-owned account (EOA) address. In all cases, however, the client should
+still sign the message using the EOA's private key.
+
+#### 3.2.1.4. Session Cookies
+
+Once the client generates a SIWE-compliant plaintext message, it signs it with the user's externally-owned account private key, resulting in an [EIP-191](https://eips.ethereum.org/EIPS/eip-191)
+Signed Data Standard format message. After generating the signed message, the client will send it to the server at a `POST /auth/login` endpoint, described below. If the server
+accepts the signed message as valid, it will respond with a session cookie linked that identifies the user's authenticated session within the server. The name of the session
+cookie is not mandated by this specification, but the server MUST use a consistent name for the session cookie, and SHOULD make a best-effort at choosing a unique name, that
+will not conflict with the session cookies served by other FiatConnect-compliant APIs. It is the client's responsibility to manage session cookies between different FiatConnect
+APIs.
+
+If a client sends a session cookie when calling the `POST /auth/login` endpoint (either to refresh their current, valid session, or to generate a new session if their current one is
+expired), the server MUST ignore the provided cookie. If the authentication request is valid, the server MUST set a new value on the cookie referencing the client's new, valid session,
+and return it to the client.
+
+#### 3.2.1.5. `POST /auth/login`
+
+The `POST /auth/login` endpoint is responsible for verifying signed messages sent by clients, creating and authenticating a user's session, and returning session cookies.
+Note that both the Sign-In With Ethereum standard, and FiatConnect, support authorization for externally owned accounts (EOAs) and contract-owned accounts; The `POST /auth/login`
+endpoint MUST honor login requests for both, as described below.
+
+##### 3.2.1.5.1. Parameters
+
+###### 3.2.1.5.1.1. Request Body
+
+The *plaintext* request body for this endpoint MUST correspond to a signed SIWE message (as described above) in the [EIP-191](https://eips.ethereum.org/EIPS/eip-191)
+Signed Data Standard format, as required by the SIWE specification.
```

Comment:
So neither SIWE nor EIP-191 actually specify how to package the signed message (i.e. I'm really not sure what you are referring to as the "Signed Data Standard format"). We need to specify that here. Two options are

1. Specify that the body is the signed message data and include the signature in an `Authorization` header.
2. Package both into the body as a JSON struct with `{ message, signature }`.

In both cases, we need to specify the encodings (e.g. `base64`) and whatnot.